src/adam_test_data/main.py

```diff
@@ -0,0 +1,945 @@
+import multiprocessing as mp
+import os
+import shutil
+import sqlite3 as sql
+import subprocess
+import uuid
+from abc import ABC, abstractmethod
+from typing import Literal, Optional, Type, Union
+
+import pyarrow as pa
+import pyarrow.compute as pc
+import pyarrow.parquet as pq
+import quivr as qv
+import ray
+from adam_core.observations import SourceCatalog
+from adam_core.propagator.utils import _iterate_chunk_indices
+from adam_core.ray_cluster import initialize_use_ray
+from adam_core.time import Timestamp
+
+from .noise import generate_noise
+from .observatories import Observatory, observatory_to_sorcha_config
+from .pointings import Pointings
+from .populations import (
+    SmallBodies,
+    orbits_to_sorcha_table,
+    photometric_properties_to_sorcha_table,
+)
+
+
+class SorchaDerivedOutputs(ABC):
+
+    @abstractmethod
+    def to_source_catalog(
+        self, catalog_id: str, exposure_id: str, observatory_code: str
+    ) -> SourceCatalog:
+        """
+        Convert the Sorcha output to a SourceCatalog.
+
+        Parameters
+        ----------
+        catalog_id : str
+            The ID of the catalog.
+        exposure_id : str
+            The ID of the exposure.
+        observatory_code : str
+            The code of the observatory.
+
+        Returns
+        -------
+        source_catalog : SourceCatalog
+            The source catalog.
+        """
+        pass
+
+
+class SorchaOutputBasic(qv.Table, SorchaDerivedOutputs):
+    # TODO: It would be really nice if the basic outputs included the FieldID
```

Comment:
Am I understanding correctly that only the full output includes the `FieldID` and that is the closest thing we get to a unique exposure id? I guess it only really matters if we want to cross reference with the pointings, as the observation data should be almost identical.